plugin/rename.py

```diff
@@ -24,10 +24,67 @@
 import sublime_plugin
 
 
+BUTTON_HTML = """
+<style>
+    html {{
+        padding: 0.2rem;
+        background-color: transparent;
+    }}
+    a {{
+        display: inline;
+        line-height: 1.5rem;
+        padding-left: 0.6rem;
+        padding-right: 0.6rem;
+        border-width: 1px;
+        border-style: solid;
+        border-color: #fff4;
+        border-radius: 4px;
+        color: #cccccc;
+        background-color: color({color} min-contrast(white 6.0));
+        text-decoration: none;
+    }}
+    html.light a {{
+        border-color: #000a;
+        color: white;
+        background-color: color({color_light} min-contrast(white 6.0));
+    }}
+</style>
+<body id='lsp-button'>
+    <a href='{command}'>{label}</a>
+</body>"""
+
+DISCARD_BUTTON_HTML = BUTTON_HTML.format(
+    label="Discard",
+    command=sublime.command_url('chain', {
+        'commands': [
+            ['hide_panel', {}],
+            ['lsp_hide_rename_buttons', {}]
+        ]
+    }),
+    color="#3f3f3f",
+    color_light="#636363"
+)
+
+
 def is_range_response(result: PrepareRenameResult) -> TypeGuard[Range]:
     return 'start' in result
 
 
+def utf16_to_code_points(s: str, col: int) -> int:
```

Comment:
I don't really feel `utf_16_pos_to_code_point_pos`. Could we keep the current name? I have reworded the docstring a little bit, and I think it's clear from the docstring and from usage what the function does.

Btw, the parameter is named `col` because that's the name given in the `View.text_point_utf16` method.

Reply:
The name to me strongly suggests something that feels like charset conversion which I think is kinda confusing.

Can we do `utf16_offset_to_code_point_offset`? Or if we want to match ST more then `utf16_offset_to_code_point_offset`?

Reply:
I'm willing to give up if you insist. I'm after all nit picking a bit.

Reply:
Well, I'd say "code points" already implies that it's about the position, because there is no character encoding named "code points". If we want to have the word "position" in the name, then I would suggest `utf16_to_utf32_position`.

If we want to mach ST closely, then it should probably be `code_point_utf16` or even the same name `text_point_utf16` (but the latter would be confusing I guess). The only difference between the builtin and this function here is that the builtin uses the counts from the beginning of the view's content, while this one takes the string as an explicit parameter, and it always uses `clamp=True`.

So I would prefer one of these:
- `utf16_to_code_points` (current)
- `utf16_to_utf32_position`
- `code_point_utf16`

Reply:
kinda like `utf16_to_utf32_position` but choose yourself whether to rename it or not later